font/src/ft/mod.rs

```diff
@@ -389,8 +397,7 @@ impl FreeTypeRasterizer {
         if use_initial_face {
             Ok(glyph_key.font_key)
         } else {
-            let key = self.load_face_with_glyph(glyph_key).unwrap_or(glyph_key.font_key);
-            Ok(key)
+            Ok(self.load_face_with_glyph(glyph_key).unwrap_or(glyph_key.font_key))
         }
     }
```

Comment:
Honestly, this entire method is a bloody mess. It has a `has_recursed` parameter, without ever being able to recurse, which is always set to `false` when called.

Then it branches two times, the first time just sets a variable to the status of that branching operation, while the second time checks that to branch on the same thing again.

Would you mind removing the parameter, changing it to `if index != 0 { return Ok(glyph_key.font_key); }` and then just putting the `Ok(self.load_face_with_glyph(glyph_key).unwrap_or(glyph_key.font_key))` at the end of it without any additional conditionals? Can get rid of the pointless `else` too.

Reply:
I've done it on refactoring PR, let me do this here one more time.